src/lib/helpers/generateServiceInterface.ts

```diff
@@ -45,15 +50,97 @@ export const generateAquaInterfaceForService = async ({
     `Failed to generate service interface for ${pathToFacadeWasm}`,
   );
 
-  const declarations = interfaceDeclaration.split(SERVICE);
-  const [serviceName, ...rest] = declarations.pop()?.split(":") ?? [];
+  return interfaceDeclaration;
+};
+
+const getServiceIdFromServiceInterface = (serviceDefinition: string) => {
+  const serviceId = serviceDefinition.split('"')[1];
+
+  assert(
+    serviceId !== undefined,
+    `Failed to parse service id from service definition: ${serviceDefinition}`,
+  );
 
-  if (serviceName === undefined || rest.length === 0) {
-    throw new Error(
-      `Failed to generate service interface for ${pathToFacadeWasm}`,
+  return serviceId;
+};
+
+export const updateAquaServiceInterfaceFile = async (
+  serviceNamePathToFacadeMap: Record<string, string>,
+  servicesFromFluenceConfig: FluenceConfig["services"],
+  marineCli: MarineCLI,
+) => {
+  let previouslyGeneratedInterfacesStr = "";
+
+  try {
+    previouslyGeneratedInterfacesStr = await readFile(
+      await ensureFluenceAquaServicesPath(),
+      FS_OPTIONS,
     );
-  }
+  } catch {}
+
+  const previouslyGeneratedInterfaces = previouslyGeneratedInterfacesStr
+    .trim()
+    .split(SERVICE_DEFINITION_SEPARATOR);
+
+  const serviceNamesFromFluenceConfig = new Set(
+    Object.keys(servicesFromFluenceConfig ?? {}),
+  );
+
+  const previouslyGeneratedInterfacesWithIdsPresentInFluenceConfig =
+    previouslyGeneratedInterfaces
+      .map((serviceDefinition) => {
+        return {
+          serviceId: getServiceIdFromServiceInterface(serviceDefinition),
+          serviceDefinition,
+        };
+      })
+      .filter(({ serviceId }) => {
+        return serviceNamesFromFluenceConfig.has(serviceId);
+      });
+
+  const setOfServiceIdsFromPreviouslyGeneratedInterfaces = new Set(
```

Comment:
I don't like this naming. I think that typescript's job is to pass info about variable type and we needn't to duplicate type info into the variable name. Probably better naming is `serviceIdsFromPreviouslyGeneratedInterfaces`